components/src/core/components/ListTable/ListTable.vue

```diff
@@ -0,0 +1,277 @@
+    },
+    flashing: {
+      type: Boolean,
+      default: false,
+    },
+    tableId: {
+      type: String,
+      default: () => nanoid(8),
+    },
+    items: {
+      type: Array as PropType<any>,
+      default: () => [],
+    },
+    selected: {
+      type: Array as PropType<number[]>,
+      default: () => [],
+    },
+    selector: {
+      type: Object as PropType<CardSelector>,
+      default: () => ({}),
+    },
+    headers: {
+      type: Array as PropType<CardHeaders>,
+      default: () => [],
+    },
+    order: {
+      type: Object as PropType<SortDefinition>,
+      default: () => ({}),
+    },
+  },
+
+  setup(props, context) {
+    provide('tableProps', readonly(props));
+    let selected = [...props.selected];
+    const flashIndexes = props.flashing ? useFlashing(props) : [];
+
+    emitter.on(`${props.tableId}-datatable:rowSelected`, value => {
+      selected = [...selected, value];
+      context.emit('update:selected', selected);
+    });
+    emitter.on(`${props.tableId}-datatable:rowUnselected`, value => {
+      selected = selected.filter(item => item !== value);
+      context.emit('update:selected', selected);
+    });
+
+    const getCheckIcon = (
+      selectedItems: Array<number>,
+      allItems: Array<RowItem>,
+    ) => {
+      return allItems.filter(
+        (item: RowItem) =>
+          item?.isSelectable !== false && item?.isDisabled !== true,
+      ).length === selectedItems.length
+        ? 'oxd-check'
+        : 'dash';
+    };
+
+    const selectAll = computed({
+      get: () => {
+        return props.selected.length > 0;
+      },
+      set: value => {
+        if (value) {
+          emitter.emit(`${props.tableId}-datatable:selectAll`);
+        } else {
+          emitter.emit(`${props.tableId}-datatable:unselectAll`);
+        }
+      },
+    });
+
+    const cardHeaders = computed(() => {
+      if (props.selectable) {
+        return [
+          {
+            name: 'selector',
+            cellType: 'oxd-table-cell-checkbox',
+          },
+          ...props.headers,
+        ];
+      }
+      return props.headers;
+    });
+
+    const checkIcon = computed(() => getCheckIcon(props.selected, props.items));
+
+    const sortFields = computed(() => {
+      const sort = {};
+      for (let key in props.order) {
+        const orderObj = props.order[key];
+        if (typeof orderObj === 'object') {
+          sort[key] = {
+            order: orderObj.order,
+            iconAsc: orderObj?.iconAsc,
+            iconDesc: orderObj?.iconDesc,
+          };
+        } else {
+          sort[key] = {
+            order: props.order[key],
+            iconAsc: '',
+            iconDesc: '',
+          };
+        }
+      }
+      return sort;
+    });
+
+    const onOrderChange = (order: Order, column: CardHeader) => {
+      const orderFields = {};
+      for (const key in props.order) {
+        orderFields[key] = 'DEFAULT';
+      }
+      context.emit('update:order', {
```

Comment:
The emitter object used to handle events is not defined in the code, so it may cause an error when the code is executed. Make sure that the emitter object is properly defined and imported.

Reply:
`setup(props, context)` setup hooks gives us he context, which has the emit function within it. not sure what you mean by emitter object is not defined? :thinking: